syntax/decode.go

```diff
@@ -65,8 +74,19 @@ func typeDecoder(t reflect.Type) decoderFunc {
 	return newTypeDecoder(t)
 }
 
+var (
+	unmarshalerType = reflect.TypeOf(new(Unmarshaler)).Elem()
+)
+
 func newTypeDecoder(t reflect.Type) decoderFunc {
-	// Note: Does not support Marshaler, so don't need the allowAddr argument
+	// XXX Unmarshalers are only supported by value.  That is, if Foo
```

Comment:
I would suggest you adopt TODO(rlb@ipv.sx)...

Reply:
Maybe this comment could say "can only be included in structs by value"

Reply:
This is actually a broader issue than unmarshalers -- pointers to any type are not supported for marshal or unmarshal.  I've added a test file to demonstrate this and filed #152